superset/assets/src/components/ConfirmStatusChange.tsx

```diff
@@ -21,54 +21,43 @@ import * as React from 'react';
 // @ts-ignore
 import { Button, Modal } from 'react-bootstrap';
 
-type ShowCallback = (callback: (e: any) => any) => (event: any) => any;
-
+type Callback = (...args: any[]) => void;
 interface Props {
   title: string | React.ReactNode;
   description: string | React.ReactNode;
-  children: (confirm: ShowCallback) => React.ReactNode;
+  onConfirm: Callback;
+  children: (showConfirm: Callback) => React.ReactNode;
 }
 
 interface State {
+  callbackArgs: any[];
   open: boolean;
-  callback: (e: React.MouseEvent) => void;
 }
-const defaultCallback = () => { console.error('ConfirmStatusChange invoked with the default callback, please provide a function to be called on confirm'); };
 export default class ConfirmStatusChange extends React.Component<Props, State> {
 
   public state = {
-    callback: defaultCallback,
+    callbackArgs: [],
     open: false,
   };
 
-  public show: ShowCallback = (callback) => (event) => {
-    if (typeof event.preventDefault === 'function') {
-      event.preventDefault();
-
-      event = {
-        ...event,
-        currentTarget: { ...event.currentTarget },
-      };
-    }
-
+  public showConfirm = (...callbackArgs: any[]) => {
     this.setState({
-      callback: () => callback(event),
+      callbackArgs,
```

Comment:
If the callback args is supposed to be the DOM event, this will not work as intended. Events can't be referenced asynchronously. https://reactjs.org/docs/events.html#event-pooling

Reply:
You'd need to check if the arg is an Event, and call `.persist()` on it.

Reply:
Good catch, I was cloning it before. None of my current use cases rely on the dom event, but probably best to handle that to avoid future weirdness.